src/Condition.js

```diff
@@ -0,0 +1,267 @@
+ */
+const configMapper = {
+  prefix: cfg => transform(cfg),
+  infix: cfg => cfg.map(child => transform(child)),
+};
+
+/**
+ * Determines how to compose VCL based on the affix type.
+ */
+const vclComposer = {
+  prefix: op => item => `${op}(${item.toVCL()})`,
+  infix: op => items => `(${items.map(item => item.toVCL()).join(op)})`,
+};
+
+/**
+ * Boolean conditions
+ */
+const booleanMap = {
+  or: {
+    mapper: configMapper.infix,
+    vcl: vclComposer.infix(' || '),
+    express: (items, req) => items.reduce((result, item) => result || item.evaluate(req), false),
+  },
+  and: {
+    mapper: configMapper.infix,
+    vcl: vclComposer.infix(' && '),
+    express: (items, req) => items.reduce((result, item) => result && item.evaluate(req), true),
+  },
+  not: {
+    mapper: configMapper.prefix,
+    vcl: vclComposer.prefix(' !'),
+    express: (item, req) => !item.evaluate(req),
+  },
+};
+
+class BooleanCondition {
+  constructor(entry, cfg) {
+    this._entry = entry;
+    this._items = this._entry.mapper(cfg);
+  }
+
+  toVCL() {
+    return this._entry.vcl(this._items, this._entry.op);
+  }
+
+  evaluate(req) {
+    return this._entry.express(this._items, req);
+  }
+}
+
+/**
+ * PropertyCondition
+ */
+class PropertyCondition {
+  constructor(prop, op, value, name) {
+    if (op && prop.allowed_ops.indexOf(op) === -1) {
+      throw new Error(`Property ${name} does not support operation: ${op}`);
+    }
+
+    this._prop = prop;
+    this._op = op;
+    this._value = value;
+    this._name = name;
+  }
+
+  toVCL() {
+    const quote = this._prop.type === 'string' ? '"' : '';
+    let value = this._value;
+    let op = this._op;
+
+    if (quote === '"' && !op) {
+      // substring-start
+      value = `^${value}`;
+      op = '~';
+    }
+    if (!op || op === '=') {
+      // operand defaults to equal
+      op = '==';
+    }
+    const vcl = typeof this._prop.vcl === 'function' ? this._prop.vcl(this) : this._prop.vcl;
+    return `${vcl} ${op} ${quote}${value}${quote}`;
+  }
+
+  evaluate(req) {
+    if (!this._prop.express) {
+      return true;
+    }
+    const actual = this._prop.express(req, this);
+    if (!actual) {
+      return false;
+    }
+
+    const value = this._value;
+    const { type } = this._prop;
+
+    switch (this._op) {
+      case '=':
+        return actual === value;
+      case '~':
+        return !!actual.match(value);
+      case '<':
+        return actual < value;
+      case '>':
+        return actual > value;
+      default:
+        return type === 'string' ? actual.startsWith(value) : actual === value;
+    }
+  }
+
+  get name() {
+    return this._name;
+  }
+
+  get type() {
+    return this._prop.type;
+  }
+}
+
+/**
+ * Known properties
+ */
+const propertyMap = {
+  url: {
+    vcl: 'req.http.X-Full-URL',
+    express: req => `${req.protocol}://${req.headers.host}/${req.originalUrl}`,
+    type: 'string',
+    allowed_ops: '=~',
+  },
+  'url.hostname': {
```

Comment:
sorry, I was wrong :-)

for symmetry reason to [hostname](https://github.com/adobe/helix-shared/blob/master/docs/giturl.schema.md#hostname), we should keep it the `name` only,
but I would add another property `url.host` that contains both. @trieloff OK ?

Reply:
Done

Reply:
I would not offer the port as something to use in conditions because it will always be `443` in production.